test/rules/library_annotations_test.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(LibraryAnnotationsTest);
+  });
+}
+
+@reflectiveTest
+class LibraryAnnotationsTest extends LintRuleTest {
+  @override
+  bool get addMetaPackageDep => true;
+
+  @override
+  String get lintRule => 'library_annotations';
+
+  test_classDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+class C {}
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_classTypeAliasDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+abstract class C = Object with Future;
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_enumDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+enum E { one, two }
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_exportDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+export 'dart:math';
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_extensionDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+extension E on int {}
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_functionDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+void f() {}
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_genericTypedefDeclaration() async {
+    await assertDiagnostics(
+      r'''
+@pragma('dart2js:late:trust')
+typedef Fn = void Function();
+''',
+      [lint(0, 29)],
+    );
+  }
+
+  test_importDirective() async {
```

Comment:
I cannot add a test case (I think) where the library_annotations lint is fired. If a `TargetKind.library` annotation is on the first directive, then the `invalid_annotation_target` Hint will not fire, but this Lint will fire.

If such an annotation is on the first declaration, then the `invalid_annotation_target` Hint _will_ fire, but this Lint _will not_ (to avoid double reporting.

So, I've added a test indicating what I wrote above.